### PR TITLE
Vision Transformer used for binary classification on the ADNI dataset (Task 6)

### PR DESCRIPTION
**Pull request by ZezAbkin**
Hi there,

This is a pull request to merge the classification problem for ADNI (AD_NC) using a Vision Transformer. This corresponds to
task 6. 

This pull request contains the following files and folders in /topic-recognition/recognition/alzheimer-transformer:

```
dataset.py: Contains data pipeline, assisting with returning dataloaders and transforming data

modules.py: Contains Vision Transformer architecture using PyTorch classes 

train.py: Script for training model, use 'python train.py -h' in terminal to see all options 

predict.py: Script for running predictions on a single image or on an entire dataset. Returns images with classifications and confidence. Use 'python predict.py -h' to see all options.

README.md:  Project Documentation and Results

readme_resources/: Plots and other results for README.md
```

For further detail and results, see README.md

Thanks,
Henry Abkin
46421797

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:

Missing model details. Need to explain what the model structure is and how it works to show your understanding of it. Also need to explain to algorithm.

Commit messages: Good, detailed.

Code:

Uses Vision transformer
Comments mainly in the model design. Less comments in the data loader training loop and prediction file.
Good design

Functionality/Performance:
- The accuracy plot looks weird. Training accuracy is lower than the val accuracy. Also, test accuracy is a **flat line** throughout training. Also the model is underfitting as the max training accuracy is only about 70%.
- 
- No hyperparamter tuning shown.
- **Test accuracy is 0.67 and doesn't change during training**

General comments:

- Could have major bugs in the code and the model also needs to train longer.
- Also need to do some hyperparameter search which hasn't been shown in the report.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem but training irregularities -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting sometimes -1
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation could be better -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good

**Comment by shakes76** [comment]
Please remove gitignore to allow merge, does not affect grade only merge.

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks lost.

**Comment by wangzhaomxy** [comment]
Have some feedback attempts but was unsuccessful. No feedback marks lost.

**Comment by ZezAbkin** [comment]
Was there an issue with the merge? I cannot find my contribution in the topic-recognition branch.

**Comment by shakes76** [comment]
Seems the gitignore file content has been removed but the file is still present in the PR? (See the Files Changed section).

**Comment by ZezAbkin** [comment]
Whoops I thought I deleted it. Should be good to merge now! Thanks for your help Shakes.
